manifests/charts/cloudcore/templates/service_cloudcore.yaml

```diff
@@ -1,54 +1,50 @@
-{{- if and (.Values.cloudCore.service.enable) }}
+{{- if .Values.cloudCore.service.enable }}
 apiVersion: v1
 kind: Service
 metadata:
+  name: cloudcore
   {{- with .Values.cloudCore.service.annotations }}
   annotations: {{- toYaml . | nindent 4 }}
   {{- end }}
   {{- with .Values.cloudCore.labels }}
   labels: {{- toYaml . | nindent 4 }}
   {{- end }}
-  name: cloudcore
 spec:
-  {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
-  type: {{ .Values.cloudCore.service.type }}
-  {{- else }}
-  type: ClusterIP
-  {{- end }}
+  type: {{ .Values.cloudCore.service.type | default "ClusterIP" }}
   ports:
-  - port: 10000
-    targetPort: 10000
-    {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
-    nodePort: {{ .Values.cloudCore.service.cloudhubNodePort }}
-    {{- end }}
-    name: cloudhub
-  - port: 10001
-    targetPort: 10001
-    protocol: UDP
-    {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
-    nodePort: {{ .Values.cloudCore.service.cloudhubQuicNodePort }}
-    {{- end }}
-    name: cloudhub-quic
-  - port: 10002
-    targetPort: 10002
-    {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
-    nodePort: {{ .Values.cloudCore.service.cloudhubHttpsNodePort }}
-    {{- end }}
-    name: cloudhub-https
-  - port: 10003
-    targetPort: 10003
-    {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
-    nodePort: {{ .Values.cloudCore.service.cloudstreamNodePort }}
-    {{- end }}
-    name: cloudstream
-  - port: 10004
-    targetPort: 10004
-    {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
-    nodePort: {{ .Values.cloudCore.service.tunnelNodePort }}
-    {{- end }}
-    name: tunnelport
+    - port: 10000
+      targetPort: 10000
+      {{- if eq .Values.cloudCore.service.type "NodePort" }}
```

Comment:
`cloudCore.service.type` and `cloudCore.hostNetWork` are different field configurations. If the `hostNetwork` is `true`, the Service will not take effect. You could consider using `hostNetwork` to determine wheter to create the Service.